subsys/net/lib/mqtt/mqtt.c

```diff
@@ -843,17 +786,7 @@ int mqtt_init(struct mqtt_ctx *ctx, enum mqtt_app app_type)
 	(void)net_context_recv(ctx->net_ctx, mqtt_recv, K_NO_WAIT, ctx);
 
 	ctx->app_type = app_type;
-
-	switch (ctx->app_type) {
-	case MQTT_APP_PUBLISHER:
-		ctx->rcv = mqtt_publisher_parser;
-		break;
-	case MQTT_APP_SUBSCRIBER:
-		ctx->rcv = mqtt_subscriber_parser;
-		break;
-	default:
-		return -EINVAL;
-	}
+	ctx->rcv = mqtt_parser;
```

Comment:
Ah, for some reason I didn't see that this was there before you did anything -- no need to make me an author since you did the work. Thanks though!